src/consumer/fetcher.spec.js

```diff
@@ -0,0 +1,48 @@
+const seq = require('../utils/seq')
+const waitFor = require('../utils/waitFor')
+const sleep = require('../utils/sleep')
+const createFetcher = require('./fetcher')
+const createWorkerQueue = require('./workerQueue')
+const createWorker = require('./worker')
+
+describe('Fetcher', () => {
+  let fetcher, fetch, handler, nodeIds, workers, workerQueue
+
+  beforeEach(() => {
+    fetch = jest.fn(async nodeId => {
+      await sleep(50)
+      return seq(10, index => `message ${index} from node ${nodeId}`)
+    })
+
+    handler = jest.fn(async () => {
+      await sleep(50)
+    })
+
+    nodeIds = [1, 2, 3]
+
+    workers = seq(5, workerId => createWorker({ workerId, handler }))
+    workerQueue = createWorkerQueue({ workers })
+    fetcher = createFetcher({ nodeIds, fetch, workerQueue })
+  })
+
+  it('should fetch all nodes', async () => {
+    fetcher.start()
+    await fetcher.stop()
+
+    expect(fetch).toHaveBeenCalledTimes(nodeIds.length)
+    nodeIds.forEach(nodeId => {
+      expect(fetch).toHaveBeenCalledWith(nodeId)
+    })
+  })
+
+  it('should utilize all workers', async () => {
+    fetcher.start()
```

Comment:
```suggestion
    await fetcher.start()
```

Reply:
The reason I'm suggesting this is:
1. I think it highlights that the API is flawed if it returns a promise you can't await
2. If the promise is rejected, we'll have an unhandled promise rejection, which isn't very fun to debug.

Reply:
this isn't a public API. 
fetchManager.start() is only used in Runner, which awaits it until an error occurs or fetchManager.stop() is called.
but as I mentioned already, TCO is not supported, so I will have to refactor it to a loop.